backstop/tests/tree.html

```diff
@@ -0,0 +1,1305 @@
+<!--
+  Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+  See LICENSE.md in the project root for license terms and full copyright notice.
+-->
+<!DOCTYPE html>
+<html
+  lang="en-US"
+  id="theme"
+>
+  <head>
+    <meta charset="UTF-8">
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1"
+    >
+    <title>Tree | iTwinUI</title>
+    <link
+      rel="stylesheet"
+      href="../../lib/css/all.css"
+    />
+    <link
+      rel="stylesheet"
+      href="../assets/demo.css"
+    />
+    <style>
+      section[id^='demo'] {
+        width: auto;
+      }
+    </style>
+    <script src="../assets/theme.js"></script>
+    <script
+      type="module"
+      src="https://cdn.skypack.dev/@itwin/itwinui-icons-elements/chevron-right"
+    ></script>
+    <script
+      type="module"
+      src="https://cdn.skypack.dev/@itwin/itwinui-icons-elements/placeholder"
+    ></script>
+  </head>
+  <body class="iui-body">
+    <theme-button></theme-button>
+    <h1>Tree</h1>
+    <hr />
+
+    <h2>Default Tree</h2>
+    <section id="demo-default">
+      <div class="iui-tree">
```

Comment:
We should use `ul` and `li` tags with corresponding roles and `aria-*` attributes for the tree as W3 has in their documantation: https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-2/treeview-2a.html

Reply:
Using `<ul>` makes sense to me.

@VirginieGaudreault, tagging you so you can follow / suggest `aria-*` tags for accessibility.

Reply:
Side note: That link is out of date. WAI ARIA Practices 1.2 should be preferred over 1.1. https://www.w3.org/TR/wai-aria-practices-1.2/#TreeView

Luckily in this case there is no difference between 1.1 and 1.2, but sometimes there are huge changes.

When in doubt, you can also just look at the bleeding edge version at https://w3c.github.io/aria-practices/

Reply:
I see, Saally, you are using ul and li elements now but if I understand recommendations correctly you need to have sub-items inside parent node element (like in the examples Mayank linked). It also should help to avoid having `--level` variable in CSS and it will let us to add animation in React.

Reply:
I added ul sub tree and got rid of the `--level` variable but this caused a problem with keeping the checkboxes left aligned. `ul` causes the entire `iui-tree-node` to shift, where as before I was only indenting `iui-node-content`. Any ideas how to get around this to shift only the node content, not the entire node or should I not nest the subtrees and keep the flat view? 
![tree](https://user-images.githubusercontent.com/88331924/143932775-79b98045-b252-4949-a253-df6d893cbdc8.gif)

Reply:
I fixed the checkboxes to make them align on the left. I had to keep the `--level` variable to set the spacing between the checkbox and `iui-node-content`. Not sure if there would be a good way to automatically set the margin by detecting how deep it is (ul li = level 0, ui li ui li = level 2) because then we have to set a max depth, with `--level` the max depth is up to the user

Reply:
Can you also add required roles from W3?

Reply:
Added roles